recipes/libid3tag/meta.yaml

```diff
@@ -0,0 +1,52 @@
+{% set name = "libid3tag" %}
+{% set version = "0.16.2" %}
+{% set sha256 = "96198b7c8803bcda44e299615e1929a85bd5a5da02e67ebc442735bc50018190" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/tenacityteam/{{ name }}/archive/refs/tags/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win]
+  skip: True  # [osx]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - make
+    - pkg-config
+  host:
+    - gperf
+    - zlib
+  run:
+    - zlib
```

Comment:
```suggestion
```
zlib has a run_export on libzlib. If your tests need the zlib headers, then add zlib to the test requirements.

Reply:
Tried that, things break, see above: https://github.com/conda-forge/staged-recipes/pull/18892/commits/7ac01e61e513a686da82fb0b34b46e23e831320f
Test does not use zlib headers.   I too was surprised why the run_export isn't working, but there is nothing zlib-specific in the tests, so wouldn't moving the requirement to the tests mean things won't work for users?

Reply:
Correct, that's why I didn't suggest adding zlib to the test environment. I can't see the build logs for your old build now, but libzlib only ships the so versioned libraries. i.e. libz.so.1 If your linker is looking for the unversioned libz, that will cause a runtime error.

I don't know specifically why we aren't shipping the unversioned libz, so you may need to patch the build.

Reply:
The link inspector from the latest build reports:

```
   INFO (libid3tag,lib/libid3tag.so.0.16.2): Needed DSO lib/libz.so.1 found in conda-forge::libzlib-1.2.11-h166bdaf_1014
```

It looks like you are linking to a versioned libz, and this libz will be included in the run requirements because libzlib is exported from libz. My guess is that your test *does* require headers or pkgconfig from the libz package.

Reply:
@notestaff, are you a part of tenacityteam?

I think the run dependency on the zlib headers is caused by this [PUBLIC link declaration](https://github.com/tenacityteam/libid3tag/blob/3df1882b76c2f44a274af1af33497125c95679b9/CMakeLists.txt#L69). PUBLIC (vs PRIVATE) implies that linkers to libid3tag need to know about the zlib API (that the zlib API is used in the libid3tag headers). If this is the case, then zlib is required for building against libid3tag. If not, then this is bug that needs to be patched. I am not sure which is the correct solution, so discuss with upstream.